### PR TITLE
[IDEA] [CRAZY] [WIP] A Smart Resource idea

### PR DESCRIPTION
**Pull request by mfalesni**
*more info coming soon*

### CODE REVIEW

File: utils/smart_resource.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+
+    def go_to_update(self):
+        if self.EDIT_LOCATION is None:
+            raise Exception("Cannot update such type of resource!")
+        sel.force_navigate(self.EDIT_LOCATION, context=self._nav_context)
+
+    def go_to_detail(self):
+        sel.force_navigate(self.DETAIL_LOCATION, context=self._nav_context)
+
+    def create(self):
```
Comment (mfalesni) [inline utils/smart_resource.py]: Thanks for comments :) will be considered after I wake up properly after two weeks of doing nothing :smile:

Comment (psav) [inline utils/smart_resource.py]: :tada: :balloon:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

File: utils/smart_resource.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        # And generate new fields
+        for key, value in attrs.iteritems():
+            if not isinstance(value, Field):
+                new_attrs[key] = value
+                continue
+            # Store fields in this hidden key
+            new_attrs["__fields__"][key] = value
+            new_attrs[key] = property(
+                fget=_create_getter_for(key),
+                fset=_create_setter_for(key) if value.can_be_set else None)
```
Comment (seandst) [inline utils/smart_resource.py]: You could probably make a [descriptor class](https://docs.python.org/2/reference/datamodel.html#implementing-descriptors) and tighten this up a little: `new_attrs[key] = _smart_resource_property(self, key)`

File: utils/smart_resource.py (hunk 1 above)
Comment (mfalesni) [inline utils/smart_resource.py]: `the separation of the create to fit into a mold that has a pre/main/post structure` it actually is not separation, it was meant for use for additional checks/actions that are required for such kind of object. But as I am looking at this now, I would write everything slightly differently :)

**Comment by psav** [comment]
Closed, to be discussed in FW3.0

**Pull request closed without merging**